fix: improve test mode toggling

eventualbuddha opened the pull request:
Rather than using a native `confirm` call, we now use a standard React modal. This allows us to provide better feedback and to block action until it's done toggling.

![test-mode-modal](https://user-images.githubusercontent.com/1938/97039127-c8fb9600-1520-11eb-9a7e-e75085da0ecf.gif)

cc @mattroe @mcchilders

beausmith commented on src/components/ToggleTestModeButton.tsx:
```diff
@@ -0,0 +1,84 @@
+
+export interface Props {
+  isTestMode: boolean
+  togglingTestMode: boolean
```
This will require a search-and-replace, but…
```suggestion
  isTogglingTestMode: boolean
```

eventualbuddha replied:
Done.

beausmith reviewed (approved):
LGTM. 1 requested change.

eventualbuddha committed:
fix: improve test mode toggling

Rather than using a native `confirm` call, we now use a standard React modal. This allows us to provide better feedback and to block action until it's done toggling.
